universe/node/api_object.go

```diff
@@ -952,3 +963,98 @@ func (n *Node) apiUnclaimAndClearCustomisation(c *gin.Context) {
 
 	c.JSON(http.StatusAccepted, true)
 }
+
+// @Summary Get tree of objects with given object as root
+// @Schemes
+// @Description Get tree of objects with given object as root
+// @Tags objects
+// @Accept json
+// @Produce json
+// @Param query query node.apiGetObjectsTree.InQuery true "query params"
+// @Success 200 {object} node.Info
+// @Failure 400 {object} api.HTTPError
+// @Router /api/v4/objects/{object_id}/tree [get]
+func (n *Node) apiGetObjectsTree(c *gin.Context) {
+	type InQuery struct {
+		MaxDepth int `form:"max_depth"`
+	}
+	var inQuery InQuery
+
+	if err := c.ShouldBindQuery(&inQuery); err != nil {
+		err := errors.WithMessage(err, "Node: apiGetObjectsTree: failed to bind query")
+		api.AbortRequest(c, http.StatusBadRequest, "invalid_request_query", err, n.log)
+		return
+	}
+
+	if inQuery.MaxDepth == 0 {
+		inQuery.MaxDepth = 10
+	}
+
+	objectID, err := umid.Parse(c.Param("objectID"))
+	if err != nil {
+		err := errors.WithMessage(err, "Node: apiGetObjectsTree: failed to parse object umid")
+		api.AbortRequest(c, http.StatusBadRequest, "invalid_object_id", err, n.log)
+		return
+	}
+
+	object, ok := n.GetObjectFromAllObjects(objectID)
+	if !ok {
+		err := errors.Errorf("Node: apiGetObjectsTree: object not found: %s", objectID)
+		api.AbortRequest(c, http.StatusNotFound, "object_not_found", err, n.log)
+		return
+	}
+
+	all := n.GetAllObjects()
+
+	result := Info{
+		ID:       object.GetID(),
+		MaxDepth: &inQuery.MaxDepth,
+	}
+
+	objectsChildren := make(map[umid.UMID][]umid.UMID)
+
+	for _, o := range all {
```

Comment:
Why the loop over all? Isn't this already in-memory as a struct field on Objects?

Reply:
Added new method for Object: `GetChildIDs() []umid.UMID`